tools/profiling/microbenchmark/bm.py

```diff
@@ -0,0 +1,109 @@
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import multiprocessing
+import os
+import subprocess
+import sys
+
+flamegraph_dir = os.path.join(os.path.expanduser('~'), 'FlameGraph')
+
+def fnize(s):
+  out = ''
+  for c in s:
+    if c in '<>, /':
+      if len(out) and out[-1] == '_': continue
+      out += '_'
+    else:
+      out += c
+  return out
+
+os.chdir(os.path.join(os.path.dirname(sys.argv[0]), '../../..'))
+if not os.path.exists('reports'):
+  os.makedirs('reports')
+
+# index html
+index_html = """
+<html>
+<head>
+<title>Microbenchmark Results</title>
+</head>
+<body>
+"""
+
+def heading(name):
+  global index_html
+  index_html += "<h1>%s</h1>\n" % name
+
+def link(txt, tgt):
+  global index_html
+  index_html += "<p><a href=\"%s\">%s</a></p>\n" % (tgt, txt)
+
+for bm_name in sys.argv[1:]:
+  # generate latency profiles
+  heading('Latency Profiles: %s' % bm_name)
+  subprocess.check_call(
+      ['make', bm_name,
+       'CONFIG=basicprof', '-j', '%d' % multiprocessing.cpu_count()])
+  for line in subprocess.check_output(['bins/basicprof/%s' % bm_name,
+                                       '--benchmark_list_tests']).splitlines():
+    link(line, 'reports/%s.txt' % fnize(line))
+    with open('reports/%s.txt' % fnize(line), 'w') as f:
+      f.write(subprocess.check_output(['bins/basicprof/%s' % bm_name,
+                                       '--benchmark_filter=^%s$' % line]))
+      f.write('\n***********************************************************\n')
+      f.write(subprocess.check_output([
+          sys.executable, 'tools/profiling/latency_profile/profile_analyzer.py',
+          '--source', 'latency_trace.txt', '--fmt', 'simple']))
+
+  # generate flamegraphs
+  heading('Flamegraphs: %s' % bm_name)
+  subprocess.check_call(
+      ['make', bm_name,
+       'CONFIG=mutrace', '-j', '%d' % multiprocessing.cpu_count()])
+  for line in subprocess.check_output(['bins/mutrace/%s' % bm_name,
+                                       '--benchmark_list_tests']).splitlines():
+    subprocess.check_call(['sudo', 'perf', 'record', '-g', '-F', '99',
```

Comment:
Since it looks like mutrace config doesn't use frame pointers, we want to use this config for other jobs we're running perf on like sweeps? - been suggested that using frame pointers is much better than e.g., "--call-graph dwarf"

I was also suggested to use fixed event period like e.g., "-c 1000" rather than average sample times with "-F", but I'm not sure about the difference in results for flame graphs yet.